### PR TITLE
Fix and refactor example app

### PR DESCRIPTION
**Pull request by kikettas**
Refactor Example app to use coordinators and fixed some issues when a user tried to import a file with _Open with_ in the Component API

### How to test
Run the app and test everything basically.

### Merging
Automatic.

### Ticket 
#112

### CODE REVIEW

File: Example/GiniVision/ComponentAPICoordinator.swift [hunk 1]
```diff
@@ -10,84 +10,129 @@ import Foundation
...
+    }
+    
+    fileprivate let documentService: DocumentService
     fileprivate var document:GiniVisionDocument?
-    fileprivate var navigationController: UINavigationController?
-    fileprivate var tabBarController: UITabBarController?
-    fileprivate weak var analysisScreen: ComponentAPIAnalysisViewController?
-    fileprivate weak var resultsScreen: ResultTableViewController?
-    fileprivate var storyboard:UIStoryboard
+    fileprivate let giniColor = UIColor(red: 0, green: (157/255), blue: (220/255), alpha: 1)
```
[the remark this answers is not in the document]
Comment (kikettas) [reply Example/GiniVision/ComponentAPICoordinator.swift]: There is not constants in the Example app, and in my opinion would be an overkill for a custom color to be in a different file.

File: Example/GiniVision/ComponentAPICoordinator.swift [hunk 2]
```diff
@@ -126,13 +126,14 @@ final class ComponentAPICoordinator: NSObject, Coordinator {
...
         resultsScreen?.document = document
-        documentService.sendFeedback(forDocument: document)
         
         if newDocumentViewController.viewControllers.first is ComponentAPIAnalysisViewController {
             resultsScreen!.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Schließen", style: .plain, target: self, action: #selector(closeComponentAPI))
         }
         
         push(viewController: resultsScreen!, removingViewControllerOfType: ComponentAPIAnalysisViewController.self)
+        
+        documentService.sendFeedback(forDocument: document)
```
Comment (nsobadjiev) [inline Example/GiniVision/ComponentAPICoordinator.swift]: Same thing, you are sending the feedback before the use has the chance to see it and possibly change something. Can you move it to when the screen is dismissed?

File: Example/Tests/GINIComponentAPICoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+//
+//  Created by Enrique del Pozo Gómez on 11/13/17.
+//  Copyright © 2017 Gini GmbH. All rights reserved.
+//
+
+import XCTest
+@testable import GiniVision_Example
+@testable import GiniVision
+
+class GINIComponentAPICoordinator: XCTestCase {
```
Comment (nsobadjiev) [inline Example/Tests/GINIComponentAPICoordinator.swift]: Only two tests for so many changes???

Comment (kikettas) [reply Example/Tests/GINIComponentAPICoordinator.swift]: I forgot to tell you that I had not finished.

File: Example/GiniVision/ComponentAPICoordinator.swift (hunk 2 above)
Comment (kikettas) [reply Example/GiniVision/ComponentAPICoordinator.swift]: Ok, I misunderstood what you meant before.

**Pull request merged**